lib/rack/utils.rb

```diff
@@ -388,6 +388,7 @@ def byte_ranges(env, size)
 
     def get_byte_ranges(http_range, size)
       # See <http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35>
+      return nil if size.zero?
```

Comment:
I would like to prefix this line with a comment that explains the rationale of ignoring the size if it is zero.